src/language-js/print/type-parameters.js

```diff
@@ -37,10 +37,6 @@ function printTypeParameters(path, options, print, paramsKey) {
     n[paramsKey].length === 0 ||
     (n[paramsKey].length === 1 &&
       (shouldHugType(n[paramsKey][0]) ||
-        (n[paramsKey][0].type === "GenericTypeAnnotation" &&
-          shouldHugType(n[paramsKey][0].id)) ||
-        (n[paramsKey][0].type === "TSTypeReference" &&
-          shouldHugType(n[paramsKey][0].typeName)) ||
```

Comment:
`shouldHugType(n[paramsKey][0].typeName))` always is `false`, because `typeName` is of type `Identifier | TSQualifiedName`.

Reply:
The conditions were added in https://github.com/prettier/prettier/pull/1735
`typescript-eslint-parser` had very different AST format back then.